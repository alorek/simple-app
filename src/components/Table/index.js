import React, { Component } from 'react';
import { connect } from 'react-redux'
import TableHead from '../TableHead';
import TableRow from '../TableRow';

class Table extends Component {

    createRows () {
        return (this.props.tableData || []).map(row => {
            return <TableRow key = { row.id } data = { row } />
        });
    }

    render() {

        const columns = [
            'resource_type',
            'display_title',
            'media_type',
            'language',
            'viewable'
        ];

        return (
            <table className="table table-striped">
                <TableHead dataKeys = { columns } />
                <tbody> { this.createRows() } </tbody>
            </table>
        );
    }
}

function mapStateToProps(state) {
    return {
        tableData: state.table.tableData
    };
}

export default connect(mapStateToProps)(Table)
