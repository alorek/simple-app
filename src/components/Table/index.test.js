import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Table from './index';
import reducer from '../../reducers';


test('Table renders correctly', () => {

    const store = createStore(reducer);
    const dataKeys = ['resource_type'];

    const tree = renderer.create(
        <Provider store={ store }>
            <Router>
                <Table  />
            </Router>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
