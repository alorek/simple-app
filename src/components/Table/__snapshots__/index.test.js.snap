// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th
        className="sorting-header"
        onClick={[Function]}
      >
        <span
          className="pull-left"
        >
          Resource Type
        </span>
        <span
          className="pull-right glyphicon glyphicon-triangle-top"
        />
      </th>
      <th
        className="sorting-header"
        onClick={[Function]}
      >
        <span
          className="pull-left"
        >
          Display Title
        </span>
      </th>
      <th
        className="sorting-header"
        onClick={[Function]}
      >
        <span
          className="pull-left"
        >
          Media Type
        </span>
      </th>
      <th
        className="sorting-header"
        onClick={[Function]}
      >
        <span
          className="pull-left"
        >
          Language
        </span>
      </th>
      <th
        className="sorting-header"
        onClick={[Function]}
      >
        <span
          className="pull-left"
        >
          Details
        </span>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Agile
      </td>
      <td>
        Agile Development
      </td>
      <td>
        flash
      </td>
      <td>
        en-us
      </td>
      <td>
        <a
          href="/record/2166"
          onClick={[Function]}
        >
          <span>
            View
          </span>
        </a>
      </td>
    </tr>
    <tr>
      <td>
        Agile
      </td>
      <td>
        Scrum
      </td>
      <td>
        flash
      </td>
      <td>
        en-us
      </td>
      <td>
        <a
          href="/record/1401"
          onClick={[Function]}
        >
          <span>
            View
          </span>
        </a>
      </td>
    </tr>
    <tr>
      <td>
        Agile
      </td>
      <td>
        Test Driven Development
      </td>
      <td>
        pdf
      </td>
      <td>
        en-us
      </td>
      <td>
        n/a
      </td>
    </tr>
    <tr>
      <td>
        Programming
      </td>
      <td>
        My Adventures in CSS
      </td>
      <td>
        html
      </td>
      <td>
        en-us
      </td>
      <td>
        <a
          href="/record/2133"
          onClick={[Function]}
        >
          <span>
            View
          </span>
        </a>
      </td>
    </tr>
    <tr>
      <td>
        Programming
      </td>
      <td>
        My Adventures in JavaScript
      </td>
      <td>
        flash
      </td>
      <td>
        en-us
      </td>
      <td>
        <a
          href="/record/2189"
          onClick={[Function]}
        >
          <span>
            View
          </span>
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;
